File: XMLCheck/Handler/Xml5Parser.cs
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using XMLCheck.Models;

namespace XMLCheck.Handler
{
    public class Xml5Parser : BaseXmlParser, IXmlParser
    {
        public string LoaiHoSo => "XML5";

        public object Parse(string xmlContent)
        {
            var result = new List<Xml5Model>();

            // 1. Load XML (Base xử lý BOM + Parse)
            if (!Load(xmlContent))
                return result;

            // 2. Check root
            if (!IsRoot("CHITIEU_CHITIET_DIENBIENLAMSANG"))
                return result;

            // 3. Lấy danh sách CHI_TIET_DIEN_BIEN_BENH
            var nodes = Root!
                .Descendants()
                .Where(e => e.Name.LocalName == "CHI_TIET_DIEN_BIEN_BENH");

            foreach (var n in nodes)
            {
                // ===== Helper cục bộ theo node =====
                string? S5(string k) =>
                    n.Descendants()
                     .FirstOrDefault(x => x.Name.LocalName == k)
                     ?.Value
                     ?.Trim();

                int? I5(string k) =>
                    int.TryParse(S5(k), out var v) ? v : null;

                DateTime? DT5(string k)
                {
                    var raw = S5(k);
                    if (string.IsNullOrWhiteSpace(raw)) return null;

                    string[] fmts = { "yyyyMMdd", "yyyyMMddHHmm", "yyyyMMddHHmmss" };
                    return DateTime.TryParseExact(
                        raw,
                        fmts,
                        CultureInfo.InvariantCulture,
                        DateTimeStyles.None,
                        out var d
                    ) ? d : null;
                }

                var m = new Xml5Model
                {
                    MA_LK = S5("MA_LK"),
                    STT = I5("STT"),

                    DIEN_BIEN_LS = S5("DIEN_BIEN_LS"),
                    GIAI_DOAN_BENH = S5("GIAI_DOAN_BENH"),

                    HOI_CHAN = S5("HOI_CHAN"),
                    PHAU_THUAT = S5("PHAU_THUAT"),

                    THOI_DIEM_DBLS = DT5("THOI_DIEM_DBLS"),
                    NGUOI_THUC_HIEN = S5("NGUOI_THUC_HIEN"),

                    DU_PHONG = S5("DU_PHONG")
                };

                result.Add(m);
            }

            return result;
        }
    }
}

File: XMLValidator_v2/src/parsers/parsers/xml5.parser.ts
import { BaseXmlParser } from '../base-xml.parser';
import { Xml5Model } from '../../interfaces/xml-models.interface';

export class Xml5Parser extends BaseXmlParser {
  public get loaiHoSo(): string {
    return 'XML5';
  }

  public parse(xmlContent: string): Xml5Model[] {
    const result: Xml5Model[] = [];

    if (!this.load(xmlContent)) return result;
    if (!this.isRoot('CHITIEU_CHITIET_DIENBIENLAMSANG')) return result;

    const nodes = this.findDescendants(this.root, 'CHI_TIET_DIEN_BIEN_BENH');

    for (const n of nodes) {
      const m: Xml5Model = {
        MA_LK: this.getElementText(n, 'MA_LK') || undefined,
        STT: this.getElementInt(n, 'STT') ?? undefined,
        DIEN_BIEN_LS: this.getElementText(n, 'DIEN_BIEN_LS') || undefined,
        GIAI_DOAN_BENH: this.getElementText(n, 'GIAI_DOAN_BENH') || undefined,
        HOI_CHAN: this.getElementText(n, 'HOI_CHAN') || undefined,
        PHAU_THUAT: this.getElementText(n, 'PHAU_THUAT') || undefined,
        THOI_DIEM_DBLS: this.getElementDate(n, 'THOI_DIEM_DBLS') ?? undefined,
        NGUOI_THUC_HIEN: this.getElementText(n, 'NGUOI_THUC_HIEN') || undefined,
        DU_PHONG: this.getElementText(n, 'DU_PHONG') || undefined,
      };
      result.push(m);
    }

    return result;
  }
}
